redbpf/src/lib.rs

```diff
@@ -729,6 +729,10 @@ impl RelocationInfo {
         let prog = programs.get_mut(&self.target_sec_idx).ok_or(Error::Reloc)?;
         // lookup the symbol we're relocating in the symbol table
         let sym = symtab[self.sym_idx];
+        // If the reloc size is 0, there is nothing to do so we skip
```

Comment:
This bit is not correct and will break working code, we should revert it for now.

As far as I understand it's needed to workaround a codegen bug in the BPF target. We should investigate that more and ideally come up with a minimal example that triggers the bug.